website/views.py

```diff
@@ -12,12 +12,15 @@
 def welcome(request):
     for visit in Visit.objects.all():
         out_of_date(visit)
-    return render(request, "website/welcome.html",
-                  {"message": "Check the available visits on our page.",
-                   "visits_all": Visit.objects.all(),
-                   "visits_available": Visit.objects.filter(available=True),
-                   "visits_available_num": Visit.objects.filter(available=True).count(),
-                   "visits_patient": Visit.objects.filter(patient=request.user)})
+    if request.user.is_active:
+        return render(request, "website/welcome.html",
+                      {"message": "Check the available visits on our page.",
```

Comment:
Code duplication (message, render). Please consider whether you really need so many querysets here.